src/Text/Layout/Table/Primitives/ColumnModifier.hs

```diff
@@ -98,13 +99,16 @@ columnModifier pos cms colModInfo = case colModInfo of
 deriveColModInfos :: Cell a => [(LenSpec, AlignSpec)] -> [Row a] -> [ColModInfo]
 deriveColModInfos specs = zipWith ($) (fmap deriveColModInfoFromColumn specs) . transpose
 
+deriveColModInfosFromColumns :: (Foldable col, Cell a) => [(LenSpec, AlignSpec)] -> [col a] -> [ColModInfo]
+deriveColModInfosFromColumns specs = zipWith ($) (fmap deriveColModInfoFromColumn specs)
```

Comment:
With this change we can reduce duplication by defining `deriveColModInfos specs = deriveColModInfoFromColumns specs . transpose`